application/btsieve/src/settings/mod.rs

```diff
@@ -43,17 +43,12 @@ pub struct Ethereum {
 }
 
 impl Settings {
-    pub fn create<D: AsRef<OsStr>>(default_config: D) -> Result<Self, ConfigError> {
+    pub fn read<D: AsRef<OsStr>>(config_file: D) -> Result<Self, ConfigError> {
         let mut config = Config::new();
 
-        let default_config_file = Path::new(&default_config);
+        let config_file = Path::new(&config_file);
 
-        // Start off by merging in the "default" configuration file
-        config.merge(File::from(default_config_file))?;
-
-        // Add in a local configuration file
-        // This file shouldn't be checked in to git
-        config.merge(File::with_name("config/local").required(false))?;
+        config.merge(File::from(config_file))?;
```

Comment:
❓ do you still need this `.merge` call? Can't you create a `Config` directly from the file?

Reply:
Unfortunately we still need it as config-rs does not support different parsing. 

I thought about moving to toml-rs instead - which would allow deserialization from a toml string - but we decided to stay with config-rs, so I left it. 
Also, toml would require us to parse the file by ourselves.

Reply:
:question: What do you mean "parse the file by ourselves"?  https://docs.rs/toml/0.5.1/toml/ supports serde which should cover everything we want to do with it unless I'm missing something.

Reply:
I meant we would need to read the file into a string first and then let toml deserialize it into our Config.

Reply:
Serde can do `from_reader` which you can get from a file I think. No need for the string inbetween.